main.tf

```diff
@@ -180,26 +192,27 @@ data "aws_iam_policy_document" "logging" {
   }
 }
 
-resource "aws_iam_policy" "default" {
-  count = local.enabled ? 1 : 0
+resource "aws_iam_policy" "s3_access_for_sftp_users" {
+  for_each = local.enabled ? local.user_names_map : {}
 
-  name   = module.iam_label.id
-  policy = join("", data.aws_iam_policy_document.allows_s3[*].json)
+  name   = "${module.iam_label.id}-${each.value}"
+  policy = data.aws_iam_policy_document.s3_access_for_sftp_users[index(local.user_names, each.value)].json
 }
 
-resource "aws_iam_policy" "logging" {
-  count = local.enabled ? 1 : 0
+resource "aws_iam_role" "s3_access_for_sftp_users" {
+  for_each = local.enabled ? local.user_names_map : {}
 
-  name   = module.logging_label.id
-  policy = join("", data.aws_iam_policy_document.logging[*].json)
+  name = "${module.iam_label.id}-${each.value}"
+
+  assume_role_policy  = join("", data.aws_iam_policy_document.assume_role_policy[*].json)
+  managed_policy_arns = [aws_iam_policy.s3_access_for_sftp_users[index(local.user_names, each.value)].arn]
 }
 
-resource "aws_iam_role" "default" {
+resource "aws_iam_policy" "logging" {
```

Comment:
How come this uses a managed iam policy instead of an inline policy?

Reply:
It is my understanding that we don't do inline policies.

Reply:
That's not really a reason lol.